stock_reserve_rule/models/stock_reserve_rule.py

```diff
@@ -296,3 +324,88 @@ def is_greater_eq(value, other):
                     # compute how much packaging we can get
                     take = (need // pack_quantity) * pack_quantity
                     need = yield location, location_quantity, take, None, None
+
+    @api.model
+    def _get_requested_comparisons(self, rounding):
+        lower_limit = self.tolerance_lower_limit
+        upper_limit = self.tolerance_upper_limit
+        requested_eq = {
+            "lot_qty_eq_requested_qty": lambda lot_qty, need_qty: lot_qty == need_qty
+        }
+        if self.tolerance_margin == "percentage":
+            upper_limit_percent = (100 + upper_limit) / 100
+            lower_limit_percent = (100 - lower_limit) / 100
+            return {
+                # need_qty + rounding < lot_qty <=
+                # need_qty * ((100 + tolerance_upper_limit) / 100)
+                "lot_qty_gt_requested_qty": lambda lot, n: n + rounding
+                < lot
+                <= n * upper_limit_percent
+                if upper_limit > 0.0
+                else lot > n,
+                # need_qty * ((100 - tolerance_lower_limit) / 100) <=
+                # lot_qty < need_qty - rounding
+                "lot_qty_lt_requested_qty": lambda lot, n: n * lower_limit_percent
+                <= lot
+                < n - rounding
+                if lower_limit > 0.0
+                else lot < n,
+                **requested_eq,
+            }
+        elif self.tolerance_margin == "absolute":
+            return {
+                # need_qty + rounding < lot_qty =< need_qty + tolerance_upper_limit
+                "lot_qty_gt_requested_qty": lambda lot, n: n + rounding
+                < lot
+                <= n + upper_limit
+                if upper_limit > 0.0
+                else lot > n,
+                # need_qty - tolerance_lower_limit <= lot_qty < need_qty - rounding
+                "lot_qty_lt_requested_qty": lambda lot, n: n - lower_limit
+                <= lot
+                < n - rounding
+                if lower_limit > 0.0
+                else lot < n,
+                **requested_eq,
+            }
+        # Default Behavior
+        return {
+            "lot_qty_gt_requested_qty": lambda lot_qty, need_qty: lot_qty > need_qty,
+            "lot_qty_lt_requested_qty": lambda lot_qty, need_qty: lot_qty < need_qty,
```

Comment:
Don't bind the tolerance to the lot strategy as this tolerance could also apply to the `packaging` strategy. Like reserve on a location containing a pallet if you need more than 80% of a pallet.

Reply:
@jbaudoux this was done as requested, please let us know what do you think as you have more experience than us on this removal strategy. Thanks!

Reply:
@jbaudoux in a previous version we tried to implement tolerance on "Full packaging" strategy but we could not define a correct behavior due to 

- lack of use case for testing and clarity of flow
- implementation being outside the scope of the PR
- budget constraint related to customer's request

So we removed the feature. It is anyway working as intended for the new "Single lot" strategy so the code is available for further implementation if the need for tolerance on full packaging arises.